modules/installation-azure-nat-gateway.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_azure/installing-azure-private.adoc
+
+[id="installation-azure-nat-gateway_{context}"]
+= Creating an Azure NAT gateway
+
+You can use link:https://docs.microsoft.com/en-us/azure/virtual-network/nat-overview[Azure VNET network address translation (NAT)]
+to provide outbound internet access for the subnets in your cluster. When using
+a VNet setup with Azure NAT and user-defined routing configured, you can create
+a private cluster with no public endpoints. You can reference
```

Comment:
I think we need to adjust the wording in general in the document for what @abhinavdahiya was highlighting:

Private Cluster = no public endpoints
UserDefinedRouting = no load balancer with a public ip

Regarding the step by step for the NAT gateway creation, I would think that a link to Azure documentation would be enough. The risk here is that even if that works right now with the current version, the example could easily be out of date and not working.